### DISCUSSION TITLE
extra-args-pre

### DISCUSSION THREAD

**Discussion by 77maxikov** [cpp-linter/cpp-linter#55] [Ideas]
May I propose another optional input extra-args-before to enable clang-tidy use extra-arg-before commandline option?
Having problems to make treat .h files as C++ headers.

**Comment by 2bndy5** [comment]
Sure, it would be a lot like the [`extra-args` input option](https://github.com/cpp-linter/cpp-linter-action#extra-args).

Is the `extra-arg-before` CLI option new to a specific version of clang-tidy or was it always there and I just didn't use it?

**Reply by 77maxikov** [reply]
Don't know ) Just found students' commits fail linter check in gh-classroom and discovered that the reason was in clang-tidy interpreting .h as C headers by default, not C++, so any includes of standard C++ library lead to errors.  Then , the only possible way to deal with it (while not renaming .h to .hpp or so) was to use extra-arg-before option.

**Reply by 2bndy5** [reply]
I think the default C++ std is 99. But, we added the `extra-args` option to configure that.

Locally, I also get related errors about header files not being parsed as C++, but I haven't since setting the C++ std to 14 in my IDE settings.

```yml
    - uses: cpp-linter/cpp-linter-action
      with:
        extra-args: '-std=c++14'
```

**Reply by 77maxikov** [reply]
Tried to use -std=c++14 in linter settings for sample project and got error: invalid argument '-std=c++14' not allowed with 'C' [clang-diagnostic-error] for sample.h  (

**Comment by 2bndy5** [comment]
I completely forgot about this. Sorry.

Currently, this problem could be avoided by using a compilation database generated from whatever build system generator you are using. If you're not using a build system generator (like CMake, meson, bazel, Gradle, ...), then this problem currently cannot be solved because any extra args given to cpp-linter are passed to every invocation of clang-tidy.
